addons/format_lame.c

```diff
@@ -0,0 +1,522 @@
+/*
+ * Asterisk -- An open source telephony toolkit.
+ *
+ * MP3 write support using LAME
+ * added by Naveen Albert <asterisk@phreaknet.org>
+ * and sponsored by Marfox Ltd.
+ *
+ * MP3 read support using LAME
+ * added by Sperl Viktor <viktike32@gmail.com>
+ *
+ * See http://www.asterisk.org for more information about
+ * the Asterisk project. Please do not directly contact
+ * any of the maintainers of this project for assistance;
+ * the project provides a web site, mailing lists and IRC
+ * channels for your use.
+ *
+ * This program is free software, distributed under the terms of
+ * the GNU General Public License Version 2. See the LICENSE file
+ * at the top of the source tree.
+ */
+
+/*!
+ * \file
+ * \brief MP3 Format including encoder and decoder
+ * \ingroup formats
+ */
+
+/*** MODULEINFO
+	<depend>lame</depend>
+	<conflict>format_mp3</conflict>
+	<defaultenabled>no</defaultenabled>
+	<support_level>extended</support_level>
+ ***/
+
+#include "asterisk.h"
+
+#include <lame/lame.h>
+
+#include "asterisk/module.h"
+#include "asterisk/mod_format.h"
+#include "asterisk/logger.h"
+#include "asterisk/format_cache.h"
+
+#define SLIN_BUFLEN 320
+#define DECORDER_OUTLEN 32768
+#define BITS 2
+
+struct lame_private {
```

Comment:
Just like not using `ast_` I would probably avoid using `lame_` as a prefix for data structures/functions we create to avoid potential conflict with liblame itself.